File: packages/corelib/src/types/captureoperationtype.ts
import { registerEnum } from "../base";

export { };

/**
 * Operation type for data change capture (used by history / audit features).
 */
export enum CaptureOperationType {
    /** Fired before the operation; allows cancellation or modification. */
    Before = 0,
    /** Entity deletion. */
    Delete = 1,
    /** Entity insertion. */
    Insert = 2,
    /** Entity update. */
    Update = 3
}

registerEnum(CaptureOperationType, 'Serenity.CaptureOperationType');

File: src/services/RequestHandlers/IntegratedFeatures/CaptureLog/CaptureOperationType.cs
namespace Serenity;

/// <summary>
/// Contains capture logging operation types
/// </summary>
public enum CaptureOperationType
{
    /// <summary>
    /// The before record, only available for updates
    /// </summary>
    Before = 0,
    /// <summary>
    /// Delete
    /// </summary>
    Delete = 1,
    /// <summary>
    /// Insert
    /// </summary>
    Insert = 2,
    /// <summary>
    /// Update (update records have both Before and Update records)
    /// </summary>
    Update = 3
}